src/lua/zencode_w3c.lua

```diff
@@ -23,9 +23,34 @@ local ETHEREUM_ADDRESS = "ethereum_address"
 local EDDSA_PUBLIC_KEY = "eddsa_public_key"
 
 local function import_did_document(doc)
-    -- id must be always present in DID-documents
+    -- id and @context must be always present in DID-documents
     ZEN.assert(doc.id, 'Invalid DID document: id not found')
-    -- all the other fields are optional and imported as a string
+    ZEN.assert(doc['@context'], 'Invalid DID document: @context not found')
+    local did_components = strtok(doc.id, '[^:]*')
```

Comment:
also add a check for size with some plausible defaults, i.e: 
- doc.id will never be longer than 256 bytes
- doc.@context will never be longer than 512 bytes
... any other limit in size that can be applied safely